src/plotting/plot.component.tsx

```diff
@@ -112,47 +122,64 @@ const Plot = (props: PlotProps) => {
           gutter={20}
           symbolSpacer={5}
           orientation="horizontal"
-          data={datasets?.map((dataset) => {
-            return { name: dataset.name, symbol: { fill: '#e31a1c' } };
-          })}
+          data={selectedChannels
+            ?.filter((channel) => channel.options.visible)
+            .map((channel) => {
+              return { name: channel.name, symbol: { fill: '#e31a1c' } };
+            })}
         />
-        {datasets?.map((dataset) => (
-          <VictoryGroup key={dataset.name}>
-            {type === 'line' && (
-              <VictoryLine
-                style={{
-                  data: { stroke: '#e31a1c' },
-                }}
-                data={dataset.data}
-                x={XAxis}
-                y={dataset.name}
-              />
-            )}
-            {/* We render a scatter graph no matter what as otherwise line charts
-            wouldn't be able to have hover tooltips */}
-            <VictoryScatter
-              style={{
-                data: { fill: '#e31a1c' },
-              }}
-              data={dataset.data}
-              x={XAxis}
-              y={dataset.name}
-              size={type === 'line' ? 2 : 3}
-              labels={({ datum }) => {
-                const formattedXLabel = formatTooltipLabel(
-                  datum._x,
-                  XAxisSettings.scale
-                );
-                const formattedYLabel = formatTooltipLabel(
-                  datum._y,
-                  YAxesSettings.scale
-                );
-                return `(${formattedXLabel}, ${formattedYLabel})`;
-              }}
-              labelComponent={<VictoryTooltip />}
-            />
-          </VictoryGroup>
-        ))}
+        {selectedChannels.map((channel) => {
+          const currentDataset = datasets.find(
```

Comment:
I wonder, if we're structuring it that we have the selected channel config in one prop and the data in another - is it not easier/more efficient to have `datasets` just be a single data array, with objects containing fields equalling the channel names? Then, when creating the lines, we go through the list of `selectedChannels`, using the name as the key for the Y axis for each line and the same data array passed as `data`?

i.e. rather than passing `datasets={[{name: "shotnum", data: [{"time": 1, "shotnum":1}, {"time": 2, "shotnum":2}, {"time": 3, "shotnum":3}]}, {name: "activeArea", data: [{"time": 1, "activeArea":1}, {"time": 2, "activeArea":2}, {"time": 3, "activeArea":3}]}]}`, we pass in `data={[{"time": 1, "shotnum":1, "activeArea": 1}, {"time": 2, "shotnum":2, "activeArea": 2}, {"time": 3, "shotnum":3, "activeArea": 3}]}` (we reduce repeated X axis info this way)

Again, this isn't necessarily the best way to do things/worth refactoring but it's just a thought

Reply:
Possibly, I just found it easier to determine what dataset matched with what selectedChannel info by just adding a name field to the dataset, instead of having to extract this from the data itself. What do you think?

Reply:
We can keep it like this for now - we can check later on if things need changing once we do more larger scale tests

Reply:
Although in the method I propose there is no difficulty in determining the dataset - as there's only one dataset that is supplied to all the lines, and just the y axis value is adjusted